### PR TITLE
Feature Decor

### PR DESCRIPTION
**Pull request by raoulkent**
Adds abstractions Drawable and fixes Positionable

### CODE REVIEW

File: game/core/src/main/java/com/gg/game/Model/Positionable.java [hunk 1]
```diff
@@ -1,7 +1,17 @@
 package com.gg.game.Model;
 
-public interface Positionable {
-    void setPos(int x, int y);
-    void setPos(Vector position);
-    Vector getPos();
+abstract class Positionable {
+    Vector position;
```
Comment (Riddarvid) [inline game/core/src/main/java/com/gg/game/Model/Positionable.java]: Should be private

Comment (Riddarvid) [reply game/core/src/main/java/com/gg/game/Model/Positionable.java]: *protected

**Comment by pudkipz** [comment]
Should Room also implement Drawable?

File: game/core/src/main/java/com/gg/game/Model/Positionable.java [hunk 2]
```diff
@@ -1,7 +1,17 @@
 package com.gg.game.Model;
 
-public interface Positionable {
-    void setPos(int x, int y);
-    void setPos(Vector position);
-    Vector getPos();
+abstract class Positionable {
```
Comment (raoulkent) [inline game/core/src/main/java/com/gg/game/Model/Positionable.java]:
```suggestion
protected abstract class Positionable {
```

Comment (Riddarvid) [reply game/core/src/main/java/com/gg/game/Model/Positionable.java]: Sorry, I meant that the variable position should be protected, the class itself should be package-private.

**Pull request closed without merging**